frontend/src/metabase/visualizations/visualizations/PieChart/chart-definition.ts

```diff
@@ -79,6 +89,91 @@ export const PIE_CHART_DEFINITION: VisualizationDefinition = {
       title: t`Dimension`,
       showColumnSetting: true,
     }),
+    "pie.rows": {
+      section: t`Data`,
+      widget: ChartSettingOrderedSimple,
+      getHidden: (_rawSeries, settings) => settings["pie.dimension"] == null,
+      getValue: (rawSeries, settings) => {
+        return getPieRows(rawSeries, settings, (value, options) =>
+          String(formatValue(value, options)),
+        );
+      },
+      getProps: (
+        _rawSeries,
+        vizSettings: ComputedVisualizationSettings,
+        onChange,
+        _extra,
+        onChangeSettings,
+      ) => {
+        return {
+          onChangeSeriesColor: (sliceKey: string, color: string) => {
+            const pieRows = vizSettings["pie.rows"];
+            if (pieRows == null) {
+              throw Error("Missing `pie.rows` setting");
+            }
+
+            onChange(
+              pieRows.map(row => {
+                if (row.key !== sliceKey) {
+                  return row;
+                }
+                return { ...row, color, defaultColor: false };
+              }),
+            );
+          },
+          onSortEnd: (newPieRows: PieRow[]) =>
+            onChangeSettings({
+              "pie.sort_rows": false,
+              "pie.rows": newPieRows,
+            }),
+        };
+      },
+      readDependencies: [
+        "pie.dimension",
+        "pie.metric",
+        "pie.colors",
+        "pie.sort_rows",
+      ],
+    },
+    "pie.sort_rows": {
+      hidden: true,
+      getDefault: getDefaultSortRows,
+    },
+    ...seriesSetting({
```

Comment:
Wdyt replacing `seriesSettings` here with a separate use of nestedSettings for the pie chart without cartesian chart logic? If we look at the implementation of `seriesSettings`, it consists of settings related to series of cartesian charts and although it can work for not series but pie rows the usage of it does not feel right here

Reply:
Done